package/network/utils/umbim/files/lib/netifd/proto/mbim.sh

```diff
@@ -23,13 +24,24 @@ proto_mbim_init_config() {
 _proto_mbim_setup() {
 	local interface="$1"
 	local tid=2
-	local ret
 
-	local device apn pincode delay $PROTO_DEFAULT_OPTIONS
-	json_get_vars device apn pincode delay auth username password $PROTO_DEFAULT_OPTIONS
+	local device pdptype apn pincode delay auth username password $PROTO_DEFAULT_OPTIONS
+	json_get_vars device pdptype apn pincode delay auth username password $PROTO_DEFAULT_OPTIONS
 
 	[ -n "$ctl_device" ] && device=$ctl_device
 
+	pdptype=$(echo "$pdptype" | awk '{print tolower($0)}')
+	case "$pdptype" in
+		ip )
+			pdptype="ipv4"
+			;;
+		ipv4 | ipv6 | ipv4v6 | ipv4-and-ipv6 )
+			;;
+		* )
+			pdptype="default"
```

Comment:
The umibm code does not recognise default as a valid iptype, actually it falls-back to default in this case. I think that better would be to not prefix the apn in case of default iptype (so to do not depend on the unrecognised string fallback)..